src/component/mxgraph/config/StyleConfigurator.ts

```diff
@@ -286,52 +286,89 @@ export default class StyleConfigurator {
-      } else if (bpmnElement instanceof ShapeBpmnEventBasedGateway) {
-        styleValues.set(StyleIdentifier.BPMN_STYLE_INSTANTIATING, String(bpmnElement.instantiate));
-        styleValues.set(StyleIdentifier.BPMN_STYLE_EVENT_BASED_GATEWAY_KIND, String(bpmnElement.gatewayKind));
-      }
+      shapeStyleValues = this.computeShapeStyle(bpmnCell);
     } else {
-      if (bpmnElement instanceof SequenceFlow) {
-        styles.push(bpmnElement.sequenceFlowKind);
+      styles.push(...this.computeEdgeStyle(bpmnCell));
+      shapeStyleValues = new Map<string, string | number>();
+    }
+
+    const fontStyleValues = this.computeFontStyleValues(bpmnCell);
+    const labelStyleValues = this.computeLabelStyleValues(bpmnCell, labelBounds);
+
+    return [] //
+      .concat([...styles])
+      .concat([...shapeStyleValues, ...fontStyleValues, ...labelStyleValues].filter(([, v]) => v && v != 'undefined').map(([key, value]) => key + '=' + value))
+      .join(';');
+  }
+
+  private computeShapeStyle(bpmnCell: Shape): Map<string, string | number>{
+    const styleValues = new Map<string, string | number>();
+    const bpmnElement = bpmnCell.bpmnElement;
+
+    if (bpmnElement instanceof ShapeBpmnEvent) {
+      styleValues.set(StyleIdentifier.BPMN_STYLE_EVENT_KIND, bpmnElement.eventKind);
+
+      if (bpmnElement instanceof ShapeBpmnBoundaryEvent || (bpmnElement instanceof ShapeBpmnStartEvent && bpmnElement.isInterrupting !== undefined)) {
+        styleValues.set(StyleIdentifier.BPMN_STYLE_IS_INTERRUPTING, String(bpmnElement.isInterrupting));
+      }
+    } else if (bpmnElement instanceof ShapeBpmnActivity) {
+      if (bpmnElement instanceof ShapeBpmnSubProcess) {
+        styleValues.set(StyleIdentifier.BPMN_STYLE_SUB_PROCESS_KIND, bpmnElement.subProcessKind);
+      } else if (bpmnElement.kind === ShapeBpmnElementKind.TASK_RECEIVE) {
+        styleValues.set(StyleIdentifier.BPMN_STYLE_INSTANTIATING, String(bpmnElement.instantiate));
       }
-      if (bpmnElement instanceof AssociationFlow) {
-        styles.push(bpmnElement.associationDirectionKind);
+
+      const markers: ShapeBpmnMarkerKind[] = bpmnElement.markers;
+      if (markers.length > 0) {
+        styleValues.set(StyleIdentifier.BPMN_STYLE_MARKERS, markers.join(','));
       }
+    } else if (ShapeUtil.isPoolOrLane(bpmnElement.kind)) {
+      // mxgraph.mxConstants.STYLE_HORIZONTAL is for the label
+      // In BPMN, isHorizontal is for the Shape
+      styleValues.set(mxgraph.mxConstants.STYLE_HORIZONTAL, bpmnCell.isHorizontal ? '0' : '1');
+    } else if (bpmnElement instanceof ShapeBpmnEventBasedGateway) {
+      styleValues.set(StyleIdentifier.BPMN_STYLE_INSTANTIATING, String(bpmnElement.instantiate));
+      styleValues.set(StyleIdentifier.BPMN_STYLE_EVENT_BASED_GATEWAY_KIND, String(bpmnElement.gatewayKind));
     }
 
+    return styleValues;
+  }
+
+  private computeEdgeStyle(bpmnCell: Edge): string[]{
```

Comment:
❓ can we rename the parameter into `edge` for clarity?

Reply:
Done